### PR TITLE
New design

### PR DESCRIPTION
**Pull request by justasojourner**
These are the changes to the MkDocs level required to support the new navigation model which I and @sspencerwire have been working on.

## Changes

1. The site name which shows in top menu bar shortened because well versioning goes like the long title blocks the version widget
2. Nav section removed - it is now being done by .pages files in content
3. I took of the edit uri because, realistically, no-one edits content directly to GitHub, so having the edit icon there causes confusion
4. The dark/light theme switcher was the wrong way round, if the light theme is active the icon should show the dark icon to switch to dark mode and vice-versa
5. I added one Python markdown extensions which is useful and is now being used in the docs
6. Changed the favicon to the standard dark green favicon, the existing was using the large white png logo
7. Added extras to make language selection more clear and added 'en' as one of the language choices.
8. Materials for MkDocs was overriding the body text size for admonitions & collapsable blocks to a smaller font I overrode that so text now displays the same as the body text.

### CODE REVIEW

**Comment by justasojourner** [comment]
> This pull request is being automatically deployed with Vercel

Automatically deployed??

**Comment by sspencerwire** [comment]
@hbjydev @NeilHanlon : These changes need to be implemented **just before** we take the new documentation scheme live, which we are pushing for within the next 7 days. If you could review the changes and let us know if you have concerns, that would be fantastic!  Thanks very much!

**Comment by sspencerwire** [comment]
@hbjydev , @NeilHanlon and any others: These are the changes that need to be coordinated and taken live just before the new documentation goes live.

**Comment by hbjydev** [comment]
> > This pull request is being automatically deployed with Vercel
> 
> Automatically deployed??

Yeah, that's part of why we use Vercel - it automatically creates "preview" deployments on PR submission

**Comment by hbjydev** [comment]
> @hbjydev @NeilHanlon : These changes need to be implemented **just before** we take the new documentation scheme live, which we are pushing for within the next 7 days. If you could review the changes and let us know if you have concerns, that would be fantastic!  Thanks very much!
> 

I'll take a look once I get to my computer :)

**Comment by hbjydev** [comment]
@sspencerwire @justasojourner 

Just loaded up this with the `testing` docs branch on my local side. One problem with the changes on your end:

image

The contributing section -- that stuff needs to be in the Wiki, not the docs. This was agreed upon a while back, not sure if Wale made you guys aware? Basically, we have a policy that anything relating to Rocky, the distro, goes in the docs, while anything relating to Rocky, the project, goes in the Wiki.

**Comment by hbjydev** [comment]
image

Same issue with this page too

**Comment by hbjydev** [comment]
image

**Comment by hbjydev** [comment]
image

**Comment by hbjydev** [comment]
There's a couple others, but you probably get the gist

**Review by hbjydev — CHANGES_REQUESTED** [review]
See my other comments on the PR.

**Comment by wsoyinka** [comment]
@hbjydev   - Thanks.  Regarding a doc/file/guide being better suited for wiki or docs  -  let's not worry about that here or let that hold things up. Content matters will be dealt with outside of here and amongst the content creators.  
Let's just focus on the web stuff, getting us up and running and getting docs.r.o  to look  fab :-)

**Comment by hbjydev** [comment]
Alright. I'm speaking with @sspencerwire separately about the content concerns. Other than that, it looks good.

File: mkdocs.yml [hunk 1]
```diff
@@ -55,26 +45,53 @@ markdown_extensions:
...
+  - pymdownx.details
+  - pymdownx.tasklist
   - footnotes
   - def_list
-  - pymdownx.tasklist
   - meta
 
 plugins:
   - search
-  - localsearch
```
Comment (hbjydev) [inline mkdocs.yml:65]: @justasojourner Why was this removed?

File: mkdocs.yml [hunk 2]
```diff
@@ -1,21 +1,10 @@
 ---
-site_name: 'Rocky Linux Documentation'
+site_name: 'Documentation'
 site_url: 'https://docs.rockylinux.org'
 docs_dir: 'docs/docs'
 repo_url: https://github.com/rocky-linux/documentation
 repo_name: rocky-linux/documentation
-edit_uri: 'edit/main/docs/'
```
Comment (hbjydev) [inline mkdocs.yml:7]: @wsoyinka -- I know having an edit link was something you specifically wanted, so is this a thing you're happy removing?

**Review by hbjydev — CHANGES_REQUESTED** [review]

File: mkdocs.yml (hunk 2 above)
Comment (wsoyinka) [reply mkdocs.yml:7]: @hbjydev   -  Thanks !!   Edit links are important.  We don't want this removed !!!!!!

Comment (hbjydev) [reply mkdocs.yml:7]: @justasojourner can you re-introduce this line again? Thanks.

**Review by hbjydev — CHANGES_REQUESTED** [review]

File: mkdocs.yml (hunk 1 above)
Comment (hbjydev) [reply mkdocs.yml:65]: @wsoyinka This too -- surely removing search isn't wanted

**Review by hbjydev — CHANGES_REQUESTED** [review]

File: mkdocs.yml [hunk 3]
```diff
@@ -1,21 +1,10 @@
 ---
-site_name: 'Rocky Linux Documentation'
+site_name: 'Documentation'
 site_url: 'https://docs.rockylinux.org'
 docs_dir: 'docs/docs'
```
Comment (wsoyinka) [inline mkdocs.yml:4]: If there is no technical reason why we have docs/docs, would like to reduce this to :

docs_dir: 'docs'

**Comment by wsoyinka** [comment]
@hbjydev   - You are absolutely correct.  We do NOT want to remove search functionality.

File: mkdocs.yml (hunk 3 above)
Comment (hbjydev) [reply mkdocs.yml:4]: This isn't something user-facing -- it sets the folder for the documentation root. So there _is_ still a technical reason for that one

**Comment by wsoyinka** [comment]
Hey @hbjydev   - You've correctly caught 2 important issues with this PR.   I  personally haven't looked through this "specific" PR in detail but from the stuff Hayden is flagging - I think we need a very specific review to go over what changes this PR will bring. 

One important item that I am sure will be touched on by this PR is the matter of versions (mike vs mkdocs).  In the experimental version of this reorg. that I reviewed, I pointed out that Rocky docs currently only has one official version.  And that we should not be presenting a version 8.3 and 8.4  on the new docs.r.o  - just because.   We will also need a notion of "latest" version.  If this PR #24  does not address any of this (mike vs mkdocs) - please disregard this last paragraph.

File: mkdocs.yml (hunk 1 above)
Comment (justasojourner) [reply mkdocs.yml:65]: This is **localsearch** (on the local computer) not website search. Search works fine without it, in fact one of the reasons I removed it was because looking at dev window in production site it can be seen that the website cannot locate the relevant JavaScript. So it fails / does not load.

File: mkdocs.yml (hunk 3 above)
Comment (wsoyinka) [reply mkdocs.yml:4]: @hbjydev   -  Yes, you are correct, it isn't user facing.  
We have successfully implemented the current setup without docs/docs .  A  single docs is simpler and cleaner.   What is the technical reason why the github docs. repo can't be cloned into/under  a single docs ?

File: mkdocs.yml (hunk 2 above)
Comment (justasojourner) [reply mkdocs.yml:7]: I _think_ I spoke about this with Steven. This is actually directly editing content ‘in line’ that is **directly editing the GitHub source***. Perhaps you were not aware of that? 

The workflow process for making changes to content is to do on a forked copy of repo on an author’s machine and then submit a PR, even those people who have authority on the repo, e.g. Steven @sspencerwire don’t edit directly. 

Go to any page on the production website (**while you are logged out of GitHub**) and click on the pencil edit icon and see what comes up. That error is what everyone sees - and given that even those who could, don’t, why have the edit button there to confuse everyone?

**Comment by justasojourner** [comment]
> Hey @hbjydev - You've correctly caught 2 important issues with this PR. I personally haven't looked through this "specific" PR in detail but from the stuff Hayden is flagging - I think we need a very specific review to go over what changes this PR will bring.

@wsoyinka - please read my two comments above. I don’t believe they actually are issues.

**Comment by justasojourner** [comment]
> One important item that I am sure will be touched on by this PR is the matter of versions (mike vs mkdocs). In the experimental version of this reorg. that I reviewed, I pointed out that Rocky docs currently only has one official version. And that we should not be presenting a version 8.3 and 8.4 on the new docs.r.o - just because. We will also need a notion of "latest" version. If this PR #24 does not address any of this (mike vs mkdocs) - please disregard this last paragraph.

As per the last statement about not going live with versions at this time **everything related to versions was removed from this PR**. 

So it is nothing to concern on that point.

**Comment by justasojourner** [comment]
> @hbjydev - You are absolutely correct. We do NOT want to remove search functionality.

We are (the PR) not.

**Comment by justasojourner** [comment]
Development machine with localsearch added
Screen Shot 2021-09-16 at 01 48 36

Localsearch removed
Screen Shot 2021-09-16 at 01 46 43

Production website
Screen Shot 2021-09-16 at 01 49 02

Local MkDocs serve server with Localsearch removed - search works.
Screen Shot 2021-09-16 at 01 52 50

File: mkdocs.yml (hunk 2 above)
Comment (wsoyinka) [reply mkdocs.yml:7]: @justasojourner   - Thanks for clarifying this point.   All this while I had been browsing the docs.r.o site, I always had an authenticated github session going and so I never noticed the undesirable behaviour you just pointed out.   Your careful attention is much appreciated :-).

Comment (justasojourner) [reply mkdocs.yml:7]: @wsoyinka no problemo - so you will advise @hbjydev that this option is not needed/desired so my change removing the edit URI (which will remove the pencil edit icon) will stand.

What about the other topic re: localsearch: 
* which is not needed
* removing of which has no impact on searching
* is not being loaded anyway

You will advise Hayden that that change stands also?

In which case there is no road-blocks for the PR.

File: mkdocs.yml (hunk 3 above)
Comment (justasojourner) [reply mkdocs.yml:4]: @wsoyinka on the point of doc/docs I would agree with @hbjydev, it has no bearing impact to documentation authors/editors (they don't even see it) so no issue to them.

Part of this is down to the fact that there are two repos - let's call them outer and inner. The outer is where the server part `mkdocs.yml` lives in and the inner is the documentation site. Now the documentation site, if one looks at the documentation repo, has its root with various stuff one of which is the folder for documentation source MD files.

The outer has to have the inner as a containing folder, it was called docs. So, from `mkdocs.yml` to get to doc source files it has to navigate to the inner root (called docs) and then to the docs folder in it.

So basically, again the authors never see all this so it's not really a concern. I think it's fine to leave it as it is.

**Comment by justasojourner** [comment]
@hbjydev / @wsoyinka 
Following the various discussions I have marked the two change requests as resolved = not needing to be changed. Please advise if any issue with that.

**Comment by hbjydev** [comment]
@sspencerwire @justasojourner @wsoyinka  Merged. Nothing should be blocking your content changes now.
